server/lib/roles/addUserRoles.ts

```diff
@@ -0,0 +1,24 @@
+import type { IRole, IUser } from '../../../definition/IUser';
+import type { IRoom } from '../../../definition/IRoom';
+import { Users, Roles } from '../../../app/models/server/raw';
+import { validateRoleList } from './validateRoleList';
+
+export const addUserRolesAsync = async (userId: IUser['_id'], roleIds: IRole['_id'][], scope?: IRoom['_id']): Promise<boolean> => {
+	if (!userId || !roleIds?.length) {
+		return false;
+	}
+
+	const user = await Users.findOneById(userId, { projection: { _id: 1 } });
+	if (!user) {
+		throw new Error('error-invalid-user');
```

Comment:
Maybe we should throw `MeteorError`s instead to assure they are shown in the UI.